lib/train_vehicles/supervisor.ex

```diff
@@ -0,0 +1,39 @@
+defmodule TrainVehicles.Supervisor do
```

Comment:
Right now we have data going from the GenState pipeline, into a pubsub library, and then into our own homegrown pubsub, which seems a bit redundant. Could we remove one of those steps? A few ideas for ways we might do that:
* Call a `GenServer.cast` directly, like `DataStatusPubSub.update()`, instead of using the pubsub library.
* Make `TrainVehiclesPubSub` a GenStage consumer so it gets the data via the genstage instead of as a separate pubsub step. I think `Realtime` isn't set up like this because the broadcasting to channels predates the genstage pipeline.
* Use the pubsub library for TrainVehiclesPubSub instead of making our own pubsub with registry.

Reply:
The benefit of this approach is that streaming from the API and sending data to a web client over a channel are decoupled. The former doesn't need to know anything about the latter and we have flexibility to do things like have something new subscribe to train data without having to make any changes.